bfrec

```diff
@@ -251,6 +283,18 @@ kernel_bootctl_update()
         exit 2
     fi
 
+    # Read device id and verify BFB signature, if needed.
+    # BFB sinature verification is ignored in BlueField-1
+    # devices.
+    device_id=$(bfhcafw mcra 0xf0014.0:16)
+    if [ "$device_id" = "0x00000214" ]; then
+        # Verify BFB for BlueField-2 device.
+        verify_bfb_signature $bfb_file 1
+    elif [ "$device_id" = "0x0000021c" ]; then
+        # Verify BFB for BlueField-3 device.
+        verify_bfb_signature $bfb_file 2
+    fi
+
```

Comment:
Can we check the get_chipnum() function in bfhcafw?  It'll be more robust to tell whether it's BF1/BF2/BF3.
The 'bfhcafw mcra' might not work for livefish mode.

Reply:
Good point, Liming.